### DISCUSSION TITLE
code-server running behind a kubernetes ingress path prefix

### DISCUSSION THREAD

**Discussion by yee379** [coder/code-server#5258] [General] [answered]
Hi,

i see that there is a method to run a code-server at a top level domain and have code-server proxy the (developed) application with proxies. this works great, however, i would like to be able to run code-server at a prefix of a domain: ie

   https://my.domain.com/code-server

should load up code-server.

however, when i configure this in my kubernetes ingress, using

    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      name: code-server-prefix
    spec:
      rules:
      - host: my.domain.com
        http:
          paths:
          - path: /code-server/
            pathType: Prefix
            backend:
              service:
                name: code-server
                port:
                  name: http

all i get going to `https://my.domain.com/code-server` is a 

    401
    Unauthorized
    go home

the 'go home' points back to my tld at https://my.domain.com/

is there an existing solution to this?

**Comment by yee379** [comment, marked as answer]
nevermind, its just be being dense: i just needed a `rewrite` annotation:

    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      name: code-server-prefix
      annotations:
        nginx.ingress.kubernetes.io/rewrite-target: /$2
    spec:
      rules:
      - host: my.domain.com
        http:
          paths:
          - path: /code-server(/|$)(.*)
            pathType: Prefix
            backend:
              service:
                name: code-server
                port:
                  name: http

**Reply by libin0811** [reply]
With your Solution I can access my code-server with the url "https://my.domain.com/code-server/login" but not "https://my.domain.com/code-server",that means i should add the "/login"in my url .Do you have the same situation?thank you
